### DISCUSSION TITLE
change the coordination when i click a second time on the map

### DISCUSSION THREAD

**Discussion by Azzedine-zemmari** [community/community#134063] [Programming Help]
### Body

i have to change the coordination when the user click on another area in the map 
the coordination still freeze with the first coordination I click on 
mapboxgl.accessToken =
"..."; // Replace with your Mapbox access token
// Coordinates of the city center
var cityCenter = [-9.233699512061456, 32.29497240385899];
// Define a bounding box around the city (ensure this covers the coordinates you are using)
var bounds = [
[-9.273, 32.27], // Southwest coordinates
[-9.2, 32.32], // Northeast coordinates
];
// Create the map instance
var map = new mapboxgl.Map({
container: "map", // container ID
style: "mapbox://styles/mapbox/streets-v11", // style URL
center: cityCenter, // starting position [lng, lat]
zoom: 15, // starting zoom (increased for better visibility)
maxBounds: bounds, // Set the bounding box as max bounds
});
const cords = [
{ id: 0, lat: 32.664108, long: -9.006899, ph: 5 },
{ id: 1, lat: 32.664623, long: -9.036011, ph: 7 },
{ id: 2, lat: 32.664128, long: -9.006222, ph: 4 },
{ id: 3, lat: 32.664148, long: -9.006859, ph: 2 },
{ id: 4, lat: 32.664128, long: -9.0332869, ph: 1 },
{ id: 5, lat: 32.632208, long: -9.03299, ph: 5 },
];
//The idea is that smaller sums indicate closer points.
map.on('click', function (e) {
const sum = [];
const lngLat = e.lngLat;
console.log("lngLat :", lngLat)
cords.forEach((cord, index) => {
    const lat = cord.lat - lngLat.lat;
    const long = cord.long - lngLat.lng;
    sum.push({ id: index, sum: lat + long });
});
console.log(sum);
//get the closer coordination and put it in the variable first
let first = sum[0];
for (let i = 0; i < sum.length; i++) {
    if (sum[i].sum < first.sum) {
        first = sum[i];
    }
}
console.log(first);
const closer = cords.find((cord) => cord.id == first.id);
let lat = document.getElementById('lat');
let long = document.getElementById('long');
let ph = document.getElementById('ph');
lat.innerHTML = closer.lat;
long.innerHTML = closer.long;
ph.innerHTML = closer.ph;

var marker = new mapboxgl.Marker()
    .setLngLat([closer.long, closer.lat])
    .addTo(map);
console.log(lat, long, ph);
console.log('closer : ', closer);
});

// Add navigation control (zoom buttons)
map.addControl(new mapboxgl.NavigationControl());
// Add geolocation control
map.addControl(
new mapboxgl.GeolocateControl({
    positionOptions: {
        enableHighAccuracy: true,
    },
    trackUserLocation: true,
})
);
this is  my js code and this HTML for  it 
<!DOCTYPE html>
<html>

<head>
    <meta charset="utf-8" />
    <title>Mapbox Example</title>
    <meta name="viewport" content="initial-scale=1,maximum-scale=1,user-scalable=no" />
    <script src="https://api.mapbox.com/mapbox-gl-js/v2.14.1/mapbox-gl.js"></script>
    <link href="https://api.mapbox.com/mapbox-gl-js/v2.14.1/mapbox-gl.css" rel="stylesheet" />
    <style>
        body {
            margin: 0;
            padding: 0;
        }

        .map {
            flex: 2;
            height: 400px;
        }

        .CarteGraphyic {
            display: flex;
            flex-direction: row;
            flex-wrap: nowrap;
        }

        .details {
            flex: 1;
        }
    </style>
</head>

<body>
    <section id="CarteGraphyic">
        <div id="map" class="map"></div>
        <div class="details">
            <h3>Latitude :</h3>
            <div id="lat"></div>
            <h3>Longtitude :</h3>
            <div id="long"></div>
            <h3>PH :</h3>
            <div id="ph"></div>
        </div>
    </section>
    <script src="/map.js"></script>
</body>

</html>
what I have to do I think about using ajax but it need a serveur

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by harisdev-netizen** [comment]
`mapboxgl.accessToken = "..."; // Replace with your Mapbox access token

// Coordinates of the city center
var cityCenter = [-9.233699512061456, 32.29497240385899];

// Define a bounding box around the city
var bounds = [
  [-9.273, 32.27], // Southwest coordinates
  [-9.2, 32.32], // Northeast coordinates
];

// Create the map instance
var map = new mapboxgl.Map({
  container: "map",
  style: "mapbox://styles/mapbox/streets-v11",
  center: cityCenter,
  zoom: 15,
  maxBounds: bounds,
});

const cords = [
  { id: 0, lat: 32.664108, long: -9.006899, ph: 5 },
  { id: 1, lat: 32.664623, long: -9.036011, ph: 7 },
  { id: 2, lat: 32.664128, long: -9.006222, ph: 4 },
  { id: 3, lat: 32.664148, long: -9.006859, ph: 2 },
  { id: 4, lat: 32.664128, long: -9.0332869, ph: 1 },
  { id: 5, lat: 32.632208, long: -9.03299, ph: 5 },
];

// Create a marker variable outside the click event
let marker;

map.on('click', function (e) {
  const sum = [];
  const lngLat = e.lngLat;
  console.log("lngLat :", lngLat);

  cords.forEach((cord, index) => {
    const lat = Math.abs(cord.lat - lngLat.lat);
    const long = Math.abs(cord.long - lngLat.lng);
    sum.push({ id: index, sum: lat + long });
  });

  console.log(sum);

  // Get the closer coordination
  let closest = sum.reduce((prev, curr) => prev.sum < curr.sum ? prev : curr);
  console.log(closest);

  const closer = cords.find((cord) => cord.id == closest.id);

  let lat = document.getElementById('lat');
  let long = document.getElementById('long');
  let ph = document.getElementById('ph');

  lat.innerHTML = closer.lat;
  long.innerHTML = closer.long;
  ph.innerHTML = closer.ph;

  // Remove the previous marker if it exists
  if (marker) {
    marker.remove();
  }

  // Create a new marker at the clicked location
  marker = new mapboxgl.Marker()
    .setLngLat([closer.long, closer.lat])
    .addTo(map);

  console.log(lat, long, ph);
  console.log('closer : ', closer);
});

// Add navigation control (zoom buttons)
map.addControl(new mapboxgl.NavigationControl());

// Add geolocation control
map.addControl(
  new mapboxgl.GeolocateControl({
    positionOptions: {
      enableHighAccuracy: true,
    },
    trackUserLocation: true,
  })
);`

If you want to test this, you can simply update your existing **map.js** file with this new code. Make sure your HTML file is correctly linked to this JavaScript file, and it should work as expected.

**Reply by Azzedine-zemmari** [reply]
thanks for your answer but it doesn't help me it still gives the same closer coordination

**Reply by Azzedine-zemmari** [reply]
![Screenshot 2024-07-30 231521](https://github.com/user-attachments/assets/12de16c5-8914-459a-9be9-45b721bdff7e)
